### PR TITLE
refactor: remove various Clippy warnings

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Trying to remove Clippy warnings for easier future warnings elimination

## What was done?

- fixed code according to Clippy
- kept `unused variables` as this is thing of discussion

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/tests/strategy_tests/upgrade_fork_tests.rs [hunk 1]
```diff
@@ -235,7 +229,7 @@ mod tests {
         } = run_chain_for_strategy(2000, hour_in_ms, strategy.clone(), config.clone(), 15);
         {
             let drive_cache = platform.drive.cache.borrow_mut();
-            let counter = drive_cache
+            let _counter = drive_cache
```
Comment (shumkov) [inline packages/rs-drive-abci/tests/strategy_tests/upgrade_fork_tests.rs:232]: Let's keep it?

File: packages/rs-drive/src/drive/grove_operations.rs [hunk 2]
```diff
@@ -93,8 +93,8 @@ fn push_drive_operation_result_optional<T>(
     value.map_err(Error::GroveDB)
 }
 
-pub type EstimatedIntermediateFlagSizes = IntMap<u32>;
-pub type EstimatedValueSize = u32;
+pub type _EstimatedIntermediateFlagSizes = IntMap<u32>;
+pub type _EstimatedValueSize = u32;
```
Comment (shumkov) [inline packages/rs-drive/src/drive/grove_operations.rs:97]: Let's keep it

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/tests/strategy_tests/upgrade_fork_tests.rs (hunk 1 above)
Comment (antouhou) [reply packages/rs-drive-abci/tests/strategy_tests/upgrade_fork_tests.rs:232]: Yeah, that's exactly what it's doing - clippy won't be happy unless you named an unused variable with _, as by the style guide it signifies that that's a variable that's going to be used later

Comment (shumkov) [reply packages/rs-drive-abci/tests/strategy_tests/upgrade_fork_tests.rs:232]: Yeah, but I think this variable is here by mistake so I would keep this clippy warning until we fix it properly.

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**